config/base/derive/src/configurable.rs

```diff
@@ -0,0 +1,228 @@
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{Attribute, Ident, Type, TypePath};
+
+use super::utils::{
+    extract_field_attrs, extract_field_idents, extract_field_types, field_has_as_str_attr,
+    field_has_inner_attr, field_has_option_type, gen_field_env, gen_lvalues, get_env_prefix,
+    get_inner_type, StructWithFields,
+};
+
+pub(super) fn impl_configurable(ast: &StructWithFields) -> TokenStream {
```

Comment:
```suggestion
pub fn impl_configurable(ast: &StructWithFields) -> TokenStream {
```

There's absolutely no point in hiding this function.